pkg/skaffold/watch/triggers.go

```diff
@@ -180,3 +185,40 @@ func (t *fsNotifyTrigger) Start(ctx context.Context) (<-chan bool, error) {
 
 	return trigger, nil
 }
+
+type apiTrigger struct {
+	Trigger  chan bool
+	Interval time.Duration
+}
+
+// Start receives triggers from gRPC/HTTP and triggers a rebuild.
+func (t *apiTrigger) Start(ctx context.Context) (<-chan bool, error) {
+	trigger := make(chan bool)
+
+	go func() {
+		timer := time.NewTimer(1<<63 - 1) // Forever
+
+		for {
+			select {
+			case <-t.Trigger:
+				logrus.Debugln("build request received")
```

Comment:
Currently this does nothing right? The processing of this trigger will happen here.
 Are we using the same Api Servier for listening to different types of triggers like Build, Deploy?
If that is the case, the channel needs specify the trigger type

Reply:
I submitted this before reading your comment on my  initial comment.
I assume this will also deploy then?

Reply:
yep, the deploy behavior will remain the same